src/audio/dcblock/dcblock.c

```diff
@@ -65,255 +78,128 @@ static void dcblock_init_state(struct comp_data *cd)
 
 /**
  * \brief Creates DC Blocking Filter component.
- * \return Pointer to DC Blocking Filter component device.
+ * \return success.
  */
-static struct comp_dev *dcblock_new(const struct comp_driver *drv,
-				    const struct comp_ipc_config *config,
-				    const void *spec)
+static int dcblock_init(struct processing_module *mod)
 {
-	struct comp_dev *dev;
+	struct module_data *md = &mod->priv;
+	struct comp_dev *dev = mod->dev;
+	struct module_config *ipc_dcblock = &md->cfg;
 	struct comp_data *cd;
-	const struct ipc_config_process *ipc_dcblock = spec;
 	size_t bs = ipc_dcblock->size;
 	int ret;
 
-	comp_cl_info(&comp_dcblock, "dcblock_new()");
-
-	dev = comp_alloc(drv, sizeof(*dev));
-	if (!dev)
-		return NULL;
-	dev->ipc_config = *config;
+	comp_info(dev, "dcblock_init()");
 
 	cd = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM, sizeof(*cd));
-	if (!cd) {
-		rfree(dev);
-		return NULL;
-	}
-
-	comp_set_drvdata(dev, cd);
+	if (!cd)
+		return -ENOMEM;
 
+	md->private = cd;
 	cd->dcblock_func = NULL;
-	/**
-	 * Copy over the coefficients from the blob to cd->R_coeffs
-	 * Set passthrough if the size of the blob is invalid
-	 */
-	if (bs == sizeof(cd->R_coeffs)) {
-		ret = memcpy_s(cd->R_coeffs, bs, ipc_dcblock->data, bs);
-		assert(!ret);
-	} else {
-		if (bs > 0)
-			comp_cl_warn(&comp_dcblock, "dcblock_new(), binary blob size %i, expected %i",
-				     bs, sizeof(cd->R_coeffs));
-		dcblock_set_passthrough(cd);
+
+	/* component model data handler */
+	cd->model_handler = comp_data_blob_handler_new(dev);
+	if (!cd->model_handler) {
+		comp_err(dev, "dcblock_init(): comp_data_blob_handler_new() failed.");
+		ret = -ENOMEM;
+		goto err;
 	}
 
-	dev->state = COMP_STATE_READY;
-	return dev;
+	ret = comp_init_data_blob(cd->model_handler, bs, ipc_dcblock->data);
+	if (ret < 0) {
+		comp_err(dev, "dcblock_init(): comp_init_data_blob() failed with error: %d", ret);
+		comp_data_blob_handler_free(cd->model_handler);
+		goto err;
```

Comment:
would be more future-proof to add a second error-handling label and move `comp_data_blob_handler_free()` there

Reply:
Yep, done!